inst/include/dust/densities.hpp

```diff
@@ -50,12 +50,18 @@ HOSTDEVICE T dbinom(int x, int size, T prob, bool log) {
   static_assert(std::is_floating_point<T>::value,
                 "dbinom should only be used with real types");
 #endif
+  T ret;
   if (x == 0 && size == 0) {
-    return maybe_log(0, log);
+    ret = 0;
+  } else {
+    ret = lchoose<T>(size, x) +
+      x * std::log(prob) +
+      (size - x) * std::log(1 - prob);
   }
-  const T ret = lchoose<T>(size, x) +
-    x * std::log(prob) +
-    (size - x) * std::log(1 - prob);
+
+#ifdef __CUDA_ARCH__
```

Comment:
Could we put this into `inst/include/dust/cuda.cuh` so that we only see here

```
SYNCWARP
```

perhaps?